src/runtime/ObjectStructure.cpp

```diff
@@ -167,17 +180,19 @@ std::pair<size_t, Optional<const ObjectStructureItem*>> ObjectStructureWithTrans
 {
     size_t size = m_properties.size();
 
-    if (LIKELY(s.hasAtomicString() && !m_hasNonAtomicPropertyName)) {
-        for (size_t i = 0; i < size; i++) {
-            if (m_properties[i].m_propertyName.rawValue() == s.rawValue()) {
-                return std::make_pair(i, &m_properties[i]);
+    if (LIKELY(s.hasAtomicString())) {
+        if (LIKELY(!m_hasNonAtomicPropertyName)) {
+            for (size_t i = 0; i < size; i++) {
+                if (m_properties[i].m_propertyName.rawValue() == s.rawValue()) {
+                    return std::make_pair(i, &m_properties[i]);
+                }
             }
-        }
-    } else if (LIKELY(s.hasAtomicString())) {
-        AtomicString as = s.asAtomicString();
-        for (size_t i = 0; i < size; i++) {
-            if (m_properties[i].m_propertyName == as) {
-                return std::make_pair(i, &m_properties[i]);
+        } else {
+            AtomicString as = s.asAtomicString();
+            for (size_t i = 0; i < size; i++) {
+                if (m_properties[i].m_propertyName == as) {
+                    return std::make_pair(i, &m_properties[i]);
+                }
             }
```

Comment:
Why don't you add symbol case check like `ObjectStructureWithoutTransition::findProperty`?
Below is the part of that from  `ObjectStructureWithoutTransition::findProperty`.

```c++
    } else if (s.isSymbol()) {
        if (m_hasSymbolPropertyName) {
            for (size_t i = 0; i < size; i++) {
                if ((*m_properties)[i].m_propertyName == s) {
                    return std::make_pair(i, &(*m_properties)[i]);
                }
            }
        }
    } 
```

Reply:
If 's' is AtomicString, we don't needs to check 'm_hasSymbolPropertyName'
it only covers there is a symbol on structure

Reply:
I mean when `s` is a Symbol case.
if `s` is a Symbol then, we need to iterate on each property when only `m_hasSymbolPropertyName` is true like `ObjectStructureWithoutTransition::findProperty` method.
Otherwise, we could just return the failed result without iterations.

Reply:
I understood it!
I added that part but I accidentally removed that part :(

Reply:
I've updated my pr